### PR TITLE
chore: Refactored importables to be artifact agnostic

### PR DESCRIPTION
**Pull request by nidhi-nair**
Split changes for package import

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/importable/ActionCollectionImportableServiceCEImpl.java [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+    }
+
+    private Mono<ImportActionCollectionResultDTO> createImportActionCollectionMono(
+            List<ActionCollection> importedActionCollectionList,
+            Mono<? extends ImportableArtifact> importableArtifactMono,
+            ImportingMetaDTO importingMetaDTO,
+            MappedImportableResourcesDTO mappedImportableResourcesDTO) {
+
+        ArtifactBasedImportableService<ActionCollection, ?> artifactBasedExportableService =
+                getArtifactBasedExportableService(importingMetaDTO);
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/importable/ActionCollectionImportableServiceCEImpl.java:75]: let's rename the method to `getArtifactBasedImportableService`

File: app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/importable/ActionCollectionImportableServiceCEImpl.java [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+
+        Mono<List<ActionCollection>> importedActionCollectionMono = Mono.just(importedActionCollectionList);
+
+        if (importingMetaDTO.getAppendToArtifact()) {
+            importedActionCollectionMono = importedActionCollectionMono.map(importedActionCollections -> {
+                List<String> importedContextNames =
+                        artifactBasedExportableService.getImportedContextNames(mappedImportableResourcesDTO);
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/actioncollections/importable/ActionCollectionImportableServiceCEImpl.java:82]: same

File: app/server/appsmith-server/src/main/java/com/appsmith/server/jslibs/importable/CustomJSLibImportableServiceCEImpl.java [hunk 3]
```diff
@@ -23,15 +24,22 @@ public CustomJSLibImportableServiceCEImpl(CustomJSLibService customJSLibService)
         this.customJSLibService = customJSLibService;
     }
 
+    @Override
+    public ArtifactBasedImportableService<CustomJSLib, ?> getArtifactBasedExportableService(
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/jslibs/importable/CustomJSLibImportableServiceCEImpl.java:28]: let's rename all such instances of similar variables

File: app/server/appsmith-server/src/main/java/com/appsmith/server/imports/importable/ImportableServiceCE.java [hunk 4]
```diff
@@ -1,27 +1,28 @@
...
 import com.appsmith.server.dtos.ArtifactExchangeJson;
 import com.appsmith.server.dtos.ImportingMetaDTO;
 import com.appsmith.server.dtos.MappedImportableResourcesDTO;
+import com.appsmith.server.imports.importable.artifactbased.ArtifactBasedImportableService;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public interface ImportableServiceCE<T extends BaseDomain> {
 
+    ArtifactBasedImportableService<T, ?> getArtifactBasedExportableService(ImportingMetaDTO importingMetaDTO);
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/imports/importable/ImportableServiceCE.java:15]: variable renaming

**Review by sondermanish — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/imports/importable/artifactbased/ArtifactBasedImportableServiceCE.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import com.appsmith.server.domains.Context;
+import com.appsmith.server.domains.ImportableArtifact;
+import com.appsmith.server.dtos.ImportingMetaDTO;
+import com.appsmith.server.dtos.MappedImportableResourcesDTO;
+import reactor.core.publisher.Flux;
+
+import java.util.List;
+import java.util.Map;
+
+public interface ArtifactBasedImportableServiceCE<T extends BaseDomain, U extends ImportableArtifact> {
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/imports/importable/artifactbased/ArtifactBasedImportableServiceCE.java:13]: non blocking nit: please add description to these interface methods

**Review by sondermanish — APPROVED** [review]

**Comment by nidhi-nair** [comment]
Force merging since failures are unrelated to git and manual reproduction has not been possible.

**Pull request merged**